library/edi.inc

```diff
@@ -768,6 +649,459 @@ function getUsernames()
+--$boundary
+Content-Disposition: form-data; name="ProcessingMode"
+
+RealTime
+--$boundary
+Content-Disposition: form-data; name="TimeStamp"
+
+$now_date
+--$boundary
+Content-Disposition: form-data; name="PayloadID"
+
+$payloadId
+--$boundary
+Content-Disposition: form-data; name="CORERuleVersion"
+
+2.2.0
+--$boundary
+Content-Disposition: form-data; name="ReceiverID"
+
+$receiver_id
+--$boundary
+Content-Disposition: form-data; name="SenderID"
+
+$sender_id
+--$boundary
+Content-Disposition: form-data; name="PayloadType"
+
+X12_270_Request_005010X279A1
+--$boundary
+Content-Disposition: form-data; name="UserName"
+
+$rt_user
+--$boundary
+Content-Disposition: form-data; name="Password"
+
+$rt_passwrd
+--$boundary
+Content-Disposition: form-data; name="Payload"
+
+$x12_270
+--$boundary--
+MIMEBODY;
+    // send the request
+    $response = oeHttp::bodyFormat('body')
+        //->setDebug('5000')/* @todo uncomment and set proxy port to debug eg Fiddler */
+        ->usingHeaders($headers)
+        ->post('https://wsd.officeally.com/TransactionSite/rtx.aspx', $mime_body); // @TODO put request urls in x12 partner's for versatility.
+
+    $formBody = $response->body();
+    $contentType = $response->header('Content-Type')[0];
+    $hContentLength = (int)$response->header('Content-Length')[0];
+    $cksum = ($hContentLength - strlen($formBody)) === 0 ? true : false; // validate content size
+    $formData = mimeParse($formBody, $contentType);
+
+    $errors = '';
+    if (!$cksum) {
+        $errors .= xlt("Request Content Fails Integrity Test");
+    }
+    if ($response->status() !== 200) {
+        $errors .= xlt("Http Error") . ": " . $response->getReasonPhrase() .  " : " . $response->status();
+    }
+    if ($formData['ErrorCode'] != "Success") {
+        $errors .= $formData['ErrorCode'] . "\n" . $formData['ErrorMessage'];
+    }
+    if ($errors) {
+        die($errors);
+    }
+
+    $x12_271 = $formData['Payload'];
+/*$x12_271 = <<<test
+
+test;*/
+
+    return $x12_271;
+}
+
+function mimeParse($formBody = '', $contentType)
+{
+    $mimeBody = preg_replace('~\r\n?~', "\r", $formBody);
+    list($contentType, $bound, $cs) = explode(";", trim($contentType)); // $contentType & $cs are throwaways
+    $bound = explode("=", trim($bound, ' '))[1];
+    $mimeFields = preg_split("/-+$bound/", $mimeBody);
+    array_pop($mimeFields);
+    $hold = $isMatches = [];
+    foreach ($mimeFields as $id => $field) {
+        if (empty($field)) {
+            continue;
+        }
+        preg_match('/name=\"([^\"]*)\"[\n|\r]+([^\n\r].*)?\r$/s', $field, $isMatches);
+        if (preg_match('/^(.*)\[\]$/i', $isMatches[1], $hold)) {
+            $mimeData[$hold[1]][] = $isMatches[2];
+        } else {
+            $mimeData[$isMatches[1]] = $isMatches[2];
+        }
+    }
+    return $mimeData;
+}
+
+function getPatientMatch($fn, $ln, $sex, $dob)
+{
+    $fn = "%" . text($fn) . "%";
+    $ln = "%" . text($ln) . "%";
+    $sex = "%" . text($sex) . "%";
```

Comment:
Should not escape the above 3 lines since they are being used i sql query, which get escaped for that via the binding.